pkg/client/apiutil/lazyrestmapper.go

```diff
@@ -200,22 +210,18 @@ func (m *lazyRESTMapper) addKnownGroupAndReload(groupName string, versions ...st
 // findAPIGroupByName returns API group by its name.
 func (m *lazyRESTMapper) findAPIGroupByName(groupName string) (metav1.APIGroup, error) {
 	// Ensure that required info about existing API groups is received and stored in the mapper.
-	// It will make 2 API calls to /api and /apis, but only once.
-	if m.apiGroups == nil {
-		apiGroups, err := m.client.ServerGroups()
-		if err != nil {
-			return metav1.APIGroup{}, fmt.Errorf("failed to get server groups: %w", err)
-		}
-		if len(apiGroups.Groups) == 0 {
-			return metav1.APIGroup{}, fmt.Errorf("received an empty API groups list")
-		}
-
-		m.apiGroups = apiGroups
+	// It will make 2 API calls to /api and /apis respectively.
+	apiGroups, err := m.client.ServerGroups()
```

Comment:
Shouldn't we cache this and only re-fetch if the cached serverGroups don't have the requested groupName?

Reply:
Yeah, I returned caching here again. Now we just update it when nothing was found.